### ISSUE TITLE
No longer able to connect to PS4 waker - cannot detect any matching device

### ISSUE THREAD

**Issue by AdamZA1** [dhleong/ps4-waker#130]
Hello,

After some initial configuration issues, I successfully had ps4-waker working for several days with no issues. However, today for some reason, ps4-waker cannot detect the PS4. It was really difficult to get the credentials to begin with so I don't want to delete and start again unless truly necessary.

Here is the error I receive, running on MacOS in terminal.

```
[16/01/2020, 14:36:32] [PS4Waker:] Error: Could not detect any matching PS4 device
    at Detector.onTimeout (/usr/local/lib/node_modules/homebridge-ps4-waker-platform/node_modules/ps4-waker/dist/detector.js:120:16)
    at Detector.emit (events.js:317:22)
    at Detector.close (/usr/local/lib/node_modules/homebridge-ps4-waker-platform/node_modules/ps4-waker/dist/detector.js:93:8)
    at listOnTimeout (internal/timers.js:537:17)
    at processTimers (internal/timers.js:481:7)
```

Any suggestions would be very welcome.

**Comment by dhleong** [comment]
Have you followed [this troubleshooting guide](https://github.com/dhleong/ps4-waker/wiki/Troubleshooting#ps4-waker-cant-see-the-ps4)? That covers all the common cases I'm familiar with. I'm not sure why it would suddenly not work (I just checked and I can still detect mine just fine) other than:

1. PS4 system update that broke something for some reason (unlikely, but [not unprecedented](https://github.com/dhleong/ps4-waker/issues/121))
2. Network configuration change

**Comment by AdamZA1** [comment]
Firstly, many thanks for the very quick reply - it's much appreciated.

In the last few days, I've not had any PS4 updates so that can't be the cause. The network configuration also hasn't changed. I have the PS4 assigned to a static IP which was the same during the PS4-waker setup.

What's confusing is that if the router was blocking the port required for communication, wouldn't the PS4 Second Screen app also not work? The app works just fine.

I've had a look through the troubleshooting guide prior to opening the issue - I have specified the UDP ports on my router to ensure no packets are getting dropped. I do not understand how to bind to the adapter but since it was all working before anyway, I'm not sure this nor the UDP ports could be the problem.

**Comment by dhleong** [comment]
Have you tried explicitly providing the PS4's IP address via `--device`?

**Comment by AdamZA1** [comment]
Yes, I’ve tried that too. Tomorrow I will try the installation on another machine to see if it is the computer/homebridge setup at fault.

**Comment by AdamZA1** [comment]
So, I managed to isolate the issue to my desktop by trying the setup on my laptop which worked. I then restarted the network adapter on the desktop and configured the installation again and it now works. I’m not sure why it now works when it didn’t before. Is it something to do with binding to the adapter?

**Comment by dhleong** [comment]
Glad it's working!

What do you mean by "restarted the network adapter," for future reference?  It could be the network adapter is the culprit if your mac is connected to both ethernet and wifi—that's what the `--bind` flag is for. Basically the discovery process has `ps4-waker` send out messages onto the network asking the PS4 to make itself known, and telling any PS4 out there where it should reply. But, on some network configurations the PS4 can receive the message but not actually send back to the address it was given (or, the device can't receive the messages the PS4 sent, either way). So that could be it.

**Comment by AdamZA1** [comment]
Nope the iMac was only connected via WiFi to the network. Once I realised it was the iMac and not anything else, I began rule #1 of IT troubleshooting. Turn everything off and on again 😂 I disconnected from the WiFi, removed the network, turned off the WiFi, turned it back on, reconfigured the network and then ran PS4-waker check and it discovered the PS4.

Oddly, all other network functions worked fine before and during this problem, including all other smart & dumb devices on my Homebridge setup. Your guess is as good as mine. Hopefully just the iMac and not the plugin.
